snap/snapcraft.yaml

```diff
@@ -0,0 +1,216 @@
+name: docker
+version: 17.03.1-ce-0
+summary: Docker Linux container runtime
+description: Docker complements kernel namespacing with a high-level API which operates at the process level. It runs unix processes with strong guarantees of isolation and repeatability across servers.
+
+confinement: strict
+grade: stable
+
+plugs:
+  support:
+    interface: docker-support
+  privileged:
+    interface: docker-support
+    privileged-containers: true
+  docker-cli:
+    interface: docker
+slots:
+  docker-daemon:
+    interface: docker
+
+apps:
+  docker:
+    command: docker-wrapper docker
+    plugs:
+      - docker-cli
+      - network
+      - home
+
+  dockerd:
+    command: dockerd-wrapper
+    daemon: simple
+    plugs:
+      - network-bind
+      - firewall-control
+      - support
+      - privileged
+    slots:
+      - docker-daemon
+
+  # TODO determine if we can "provide" a "docker-compose" in PATH directly (since this'll show up as "docker.compose" in the user PATH)
+  compose:
+    command: docker-wrapper docker-compose
+    plugs:
+      - docker-cli
+      - network
+      - home
+  help:
+    command: help
+
+parts:
+  snappy-bins:
+    plugin: shell
+    shell: bash
+    shell-flags: ['-ex']
+    shell-command: |
+      install -d "$SNAPCRAFT_PART_INSTALL/bin"
+      install -t "$SNAPCRAFT_PART_INSTALL/bin" "$SNAPDIR"/bin/*
+    stage-packages:
+      - mount
+
+  go:
+    plugin: shell
+    shell: bash
+    shell-flags: ['-ex']
+    shell-command: |
+      apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys F1831DDAFC42E99D
+      add-apt-repository ppa:gary-wzl77/docker-golang
+      apt update
+      apt install -y golang-1.8
```

Comment:
Isn't this going to fail to build on Launchpad, since internet access is only enabled during the "fetch" stage of building there? (or has that changed?)